cargo-dist/src/init.rs

```diff
@@ -52,6 +52,92 @@ fn theme() -> dialoguer::theme::ColorfulTheme {
     }
 }
 
+/// Copy [workspace.metadata.dist] from one workspace to [dist] in another.
+fn copy_cargo_workspace_metadata_dist(
+    new_workspace: &mut toml_edit::DocumentMut,
+    workspace_toml: toml_edit::DocumentMut,
+) {
+    if let Some(dist) = workspace_toml
+        .get("workspace")
+        .and_then(|t| t.get("metadata"))
+        .and_then(|t| t.get("dist"))
+    {
+        new_workspace.insert("dist", dist.to_owned());
+    }
+}
+
+/// Remove [workspace.metadata.dist], if it exists.
+fn prune_cargo_workspace_metadata_dist(workspace: &mut toml_edit::DocumentMut) {
+    workspace
+        .get_mut("workspace")
+        .and_then(|ws| ws.get_mut("metadata"))
+        .and_then(|metadata_item| metadata_item.as_table_mut())
+        .and_then(|table| table.remove("dist"));
+}
+
+/// Create a toml-edit document set up for a cargo workspace.
+fn new_workspace() -> toml_edit::DocumentMut {
+    let mut new_workspace = toml_edit::DocumentMut::new();
+
+    // Write generic workspace config
+    let mut table = toml_edit::table();
+    if let Some(t) = table.as_table_mut() {
+        let mut array = toml_edit::Array::new();
+        array.push("cargo:.");
+        t["members"] = toml_edit::value(array);
+    }
+    new_workspace.insert("workspace", table);
+
+    new_workspace
+}
+
+fn do_migrate_from_rust_workspace() -> DistResult<()> {
+    let workspaces = config::get_project()?;
+    let root_workspace = workspaces.root_workspace();
+    let initted = has_metadata_table(root_workspace);
+
+    if root_workspace.kind != WorkspaceKind::Rust {
+        // we're not using a Rust workspace, so no migration needed.
+        return Ok(());
+    }
+
+    if !initted {
+        // the workspace hasn't been initialized, so no migration needed.
+        return Ok(());
+    }
+
+    // Load in the root workspace toml to edit and write back
+    let workspace_toml = config::load_toml(&root_workspace.manifest_path)?;
+    let mut original_workspace_toml = workspace_toml.clone();
+
+    // Generate a new workspace, then populate it using config from Cargo.toml.
+    let mut new_workspace = new_workspace();
+    copy_cargo_workspace_metadata_dist(&mut new_workspace, workspace_toml);
+
+    let workspace_toml = new_workspace;
```

Comment:
This assign probably isn't doing anything at this point?

Reply:
ah, good catch! i think it's a leftover from refactoring work. i can just use `new_workspace` instead of `workspace_toml` when writing the config file a few lines down.